### PR TITLE
deprecate heifload autorotate

### PR DESCRIPTION
**Pull request by jcupitt**
heifload autorotate is now always on, and we always delete EXIF
orientation tags. If a HEIC image fails to rotate upright, at least it's
someone else's fault.

see https://github.com/libvips/libvips/pull/1680

### CODE REVIEW

File: libvips/foreign/heifload.c [hunk 1]
```diff
@@ -913,9 +873,12 @@ vips_foreign_load_heif_read( void *data, size_t size, void *userdata )
 	gint64 result;
 
 	result = vips_source_read( heif->source, data, size );
+	/* On EOF, make a note of the file length.
+	 */
 	if( result == 0 &&
 		heif->length == -1 ) 
-		heif->length = vips_source_seek( heif->source, 0L, SEEK_CUR );
+		result = heif->length =
```
[the remark this answers is not in the document]
Comment (lovell) [reply libvips/foreign/heifload.c:880]: Yes, you're right, the problem I'm seeing is that `vips_foreign_load_heif_read` is being called by libheif with `size = 0` in the _middle_ of a file, which then "truncates" at the current seek point.

Adding an `if ( size == 0 ) return 0;` guard at the start of this function appears to protect against that problem, but it's unrelated to this change.

[the remark this answers is not in the document]
Comment (jcupitt) [reply libvips/foreign/heifload.c:880]: Wow, libheif asking for zero bytes is bizarre. You're right that it'll be interpreted as EOF. Let's add another guard.

[the remark this answers is not in the document]
Comment (jcupitt) [reply libvips/foreign/heifload.c:884]: OK, hopefully fixed with b643bd94

[the remark this answers is not in the document]
Comment (lovell) [reply libvips/foreign/heifload.c:884]: Thanks, that's fixed it. A sample image that triggers a zero-byte read is https://github.com/AOMediaCodec/av1-avif/blob/master/testFiles/Netflix/avif/cosmos_frame01000_yuv420_10bpc_bt2020_hlg_q50.avif

[the remark this answers is not in the document]
Comment (jcupitt) [reply libvips/foreign/heifload.c:884]: Great! Let's merge this.

**Pull request merged**